eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py

```diff
@@ -69,81 +99,146 @@ def get_assembly_info_and_date_ranges(all_log_files):
+                                                       cursor_factory=psycopg2.extras.RealDictCursor)
+    # See clustering jobs here: https://github.com/EBIvariation/eva-accession/blob/5f827ae8f062ae923a83c16070f6ebf08c544e31/eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/configuration/BeanNames.java#L76
+    jobs = ["CLUSTERING_FROM_MONGO_JOB", "STUDY_CLUSTERING_JOB", "PROCESS_REMAPPED_VARIANTS_WITH_RS_JOB",
+            "CLUSTER_UNCLUSTERED_VARIANTS_JOB"]
+    jobs_comma_separated = ",".join([f"'{job}'" for job in jobs])
+    query_to_get_step_start_end_times = f"""
+    SELECT DISTINCT
+      i.JOB_NAME as job_name,
+      j.JOB_EXECUTION_ID as job_id,
+      s.STEP_EXECUTION_ID as step_id,
+      s.STEP_NAME as step_name,
+      j.START_TIME::timestamp AT TIME ZONE 'Europe/London' as job_start_time,
+      j.END_TIME::timestamp AT TIME ZONE 'Europe/London' as job_end_time,
+      s.START_TIME::timestamp AT TIME ZONE 'Europe/London' as step_start_time,
+      s.END_TIME::timestamp AT TIME ZONE 'Europe/London' as step_end_time,
+      p.STRING_VAL as assembly
+    FROM 
+      BATCH_JOB_EXECUTION j
+      JOIN BATCH_STEP_EXECUTION s ON j.JOB_EXECUTION_ID = s.JOB_EXECUTION_ID
+      JOIN BATCH_JOB_INSTANCE i ON j.JOB_INSTANCE_ID = i.JOB_INSTANCE_ID
+      JOIN BATCH_JOB_EXECUTION_PARAMS p ON j.JOB_EXECUTION_ID = p.JOB_EXECUTION_ID
+    WHERE 
+      i.JOB_NAME in ({jobs_comma_separated})
+      AND p.KEY_NAME = 'assemblyAccession'
+      AND p.STRING_VAL = '{assembly_accession}'
+    """
+    if prev_release_end_timestamp_for_assembly:
+        release_end_timestamp_for_assembly_str = prev_release_end_timestamp_for_assembly.strftime('%Y-%m-%d %H:%M:%S')
+        query_to_get_step_start_end_times += f" AND j.start_time > " \
+                                             f"to_timestamp('{release_end_timestamp_for_assembly_str}', " \
+                                             f"'YYYY-MM-DD HH24:MI:SS')"
+    all_results = get_all_results_for_query(accessioning_job_tracker_handle, query_to_get_step_start_end_times)
+    if all_results:
+        for job_id in set([result["job_id"] for result in all_results]):
+            time_ranges_by_job_id[job_id] = defaultdict(dict)
+            time_ranges_by_job_id[job_id]["last_timestamp"] = \
+                max([result["job_end_time"] for result in all_results
+                     if result["job_id"] == job_id and result["job_end_time"]] +
+                    [result["job_start_time"] for result in all_results
+                     if result["job_id"] == job_id and result["job_start_time"]] +
+                    [result["step_end_time"] for result in all_results
+                     if result["job_id"] == job_id and result["step_end_time"]] +
+                    [result["step_start_time"] for result in all_results
+                     if result["job_id"] == job_id and result["step_start_time"]])
```

Comment:
See [here](https://github.com/EBIvariation/eva-accession/blob/b73d13edda9c305ee2d9dbf5a860793acd524212/eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py#L451) for the previous approach.